### PR TITLE
Tests arch arm irq vector table fix for armv6

### PR DESCRIPTION
**Pull request by ioannisg**
Increasing coverage for ARMv6 - adding the IRQ Vector Table test for Cortex-M Baseline, too.

### CODE REVIEW

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c [hunk 1]
```diff
@@ -16,8 +16,8 @@
  */
 #define _ISR_OFFSET 0
 
-#if defined(CONFIG_SOC_SERIES_NRF52X)
-/* The customized solution for nRF52X-based platforms
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+/* The customized solution for nRF5X-based platforms
```
Comment (anangl) [inline tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Same correction needed in [line 26](https://github.com/zephyrproject-rtos/zephyr/pull/19856/files#diff-c64ef6ac0266b8f4ff2c414cd2733cdeR26) (can't add a comment there).

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c [hunk 2]
```diff
@@ -152,24 +166,34 @@ void test_arm_irq_vector_table(void)
...
  */
 void rtc1_nrf_isr(void);
 void nrf_power_clock_isr(void);
-#if defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+void timer0_nrf_isr(void);
+vth __irq_vector_table _irq_vector_table[RTC1_IRQn + 1] = {
+	nrf_power_clock_isr,
+	0, 0, 0, 0, 0, 0, 0, timer0_nrf_isr, isr0, isr1, isr2, 0, 0, 0, 0, 0,
```
Comment (anangl) [inline tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: These trailing `0`s don't seem to be useful.

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c [hunk 3]
```diff
@@ -134,7 +147,8 @@ void test_arm_irq_vector_table(void)
 		      k_sem_take(&sem[2], K_NO_WAIT)), NULL);
 
 	for (int ii = 0; ii < 3; ii++) {
-#if defined(CONFIG_SOC_TI_LM3S6965_QEMU)
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) || \
+	defined(CONFIG_SOC_TI_LM3S6965_QEMU)
 		/* the QEMU does not simulate the
 		 * STIR register: this is a workaround
```
Comment (anangl) [inline tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Shouldn't this comment be updated as well?

Comment (ioannisg) [reply tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: No, ARMv6-M is well known not to have a STIR - I don't think this is a needed explanation. The TI SOC is ARMv7 but just the QEMU is incompletely emulating it, so we've added the comment.

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c (hunk 2 above)
Comment (ioannisg) [reply tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Thanks, will fix

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c (hunk 1 above)
Comment (ioannisg) [reply tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Thank, will add

File: tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c [hunk 4]
```diff
@@ -152,24 +166,33 @@ void test_arm_irq_vector_table(void)
...
+ *
+ * Note: qemu_cortex_m0 uses TIMER0 to implement system timer.
  */
 void rtc1_nrf_isr(void);
 void nrf_power_clock_isr(void);
-#if defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_SOC_SERIES_NRF51X) || defined(CONFIG_SOC_SERIES_NRF52X)
+#if defined(CONFIG_BOARD_QEMU_CORTEX_M0)
+void timer0_nrf_isr(void);
+vth __irq_vector_table _irq_vector_table[TIMER0_IRQn + 1 + 3] = {
```
Comment (anangl) [inline tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Why do we actually specify sizes of these arrays? Compilers should be able to determine those by counting the provided entries.

Comment (ioannisg) [reply tests/arch/arm/arm_irq_vector_table/src/arm_irq_vector_table.c]: Good question ;)
Totally missed that during all the reworks I've done in this test. I 'll add an extra commit to fix this

**Review by ioannisg — COMMENTED** [review]
@anangl added a commit that removes the explicit array size definitions on the const irq vector tables 
thanks!

**Review by anangl — APPROVED** [review]

**Pull request merged**